crypto/evp/p_lib.c

```diff
@@ -328,12 +334,29 @@ EVP_PKEY *EVP_PKEY_new_CMAC_key(ENGINE *e, const unsigned char *priv,
         goto err;
     }
 
+#if 0
     if (EVP_MAC_ctrl(cmctx, EVP_MAC_CTRL_SET_ENGINE, e) <= 0
         || EVP_MAC_ctrl(cmctx, EVP_MAC_CTRL_SET_CIPHER, cipher) <= 0
         || EVP_MAC_ctrl(cmctx, EVP_MAC_CTRL_SET_KEY, priv, len) <= 0) {
         EVPerr(EVP_F_EVP_PKEY_NEW_CMAC_KEY, EVP_R_KEY_SETUP_FAILED);
         goto err;
     }
+#endif
+    /* How to support engines ? */
+    params[paramsn++] =
+        OSSL_PARAM_construct_utf8_string("engine", (char *)engine_name,
+                                         strlen(engine_name) + 1, NULL);
+    params[paramsn++] =
+        OSSL_PARAM_construct_utf8_string("cipher", (char *)cipher_name,
+                                         strlen(cipher_name) + 1, NULL);
```

Comment:
This needs some thought. The cipher might be an explicitly fetched cipher in which case we should be using that specific cipher.